common/djangoapps/enrollment/api.py

```diff
@@ -225,7 +225,8 @@ def update_enrollment(user_id, course_id, mode=None, is_active=None):
         }
 
     """
-    _validate_course_mode(course_id, mode)
+    if mode is not None:
+        _validate_course_mode(course_id, mode)
```

Comment:
it doesn't because this endpoint is not used to create enrollments (which is the faulty assumption in my original comment).  the behavior of the function changed as we have both acknowledged, but i was wrong about a possible side effect of that.  Let us never speak of this matter again.